### ISSUE TITLE
Can't Register In Windows 11?

### ISSUE THREAD

**Issue by msueping** [cwollenhaupt/fpCefSharp#12]
Hi Christof,

Is there something that would be stopping the cmd file from successfully registering the dll on a Windows 11 PC? 

I've tried running it several times in admin mode and when I run the showhtml form, it always asks to locate the fpDotNet.dll. If I find it and select it, VFP complains with "OLE error code 0x80040154. OLE object is being ignored. Record number 7".

UPDATE: Never mind. I manually issued the registeration command from the command window and it said that it successfully registered. But, after doing that and then trying to run the form, VFP complains that it can find CefSharpBrowser. I had to add a SET PATH to get past that. Now my problem is that I'm getting the error "Cannot find entry point SetClrVersion in the DLL." in the PROTECTED FUNCTION SetClrVersion(lcVersion) where it issues SetClrVersionApi(lcVersion)

**Comment by dsdsdfsdfsfsdfsfsdf** [comment]
I have same problem and the solution is to set property to lCheckRuntime = .F in the CefSharpBrowser.PRG file or install https://aka.ms/vs/17/release/vc_redist.x86.exe

**Comment by msueping** [comment]
Thanks for the reply. I did install the vc-redist.x86.exe prior to your suggestion and set the lCheckRuntime to .F. Now, however, I'm getting the following error...
![Screenshot 2022-02-21 143245](https://user-images.githubusercontent.com/67658122/155016526-811c553b-cef9-4c50-b8f5-cd5966ff6b78.jpg)

**Comment by dsdsdfsdfsfsdfsfsdf** [comment]
Hi,
You must have the actual version of cefSharp installed from https://github.com/cwollenhaupt/fpCefSharp/releases

**Comment by msueping** [comment]
Forgive me, but, I don't understand. The link that you've posted above leads me to a zip file named "fpCefSharp.v97.1.61.zip" which is what I started with. I've extracted the content and see that there is a folder named "cef-bin-v97.1.61" that contains several CEFSharp dll files. The notes do not say anything about having to register any of them so I have not. 

I've run the register.cmd in admin mode and I've issued a SET DEFAULT to point to the folder into which I've extracted the zip contents. I've done the same for SET PATH.  I then try to run the showhtml form and get the error.

From looking thru cefsharpbrowser.prg, it would appear to me that it attempts to locate the folder named "cef-bin-v97.1.61" using the function named GetCefSharpPath.

What else am I missing?

**Comment by dsdsdfsdfsfsdfsfsdf** [comment]
HI,
Extract fpCefSharp.v97.1.61.zip into c:\temp\CefSharp\ directory. Then run Visual FoxPro and write in command window:
SET defa to c:\temp\CefSharp
DO form C:\Temp\CefSharp\showhtml.scx WITH "https://github.com/cwollenhaupt/fpCefSharp/releases"

**Comment by msueping** [comment]
I started over from scratch and did exactly what you mentioned above and I get the same error! This is nothing but frustration.

**Comment by dsdsdfsdfsfsdfsfsdf** [comment]
1. My Windows version:
![image](https://user-images.githubusercontent.com/99869018/155364933-1f992a43-0423-4221-a7af-282ba3208e65.png)
2. Extract zip file:
![image](https://user-images.githubusercontent.com/99869018/155365761-5d9b09f1-21cb-4afa-8764-ab6927c12cce.png)
3. Install https://aka.ms/vs/17/release/vc_redist.x86.exe
4. Run form:
![image](https://user-images.githubusercontent.com/99869018/155366170-cab9df6f-7db8-4842-84c1-c8cdb43f335b.png)
![image](https://user-images.githubusercontent.com/99869018/155367280-94805076-8c81-47a7-9f68-6e24c8921973.png)
5. See form...

**Comment by msueping** [comment]
Do I need to do step 3. Install https://aka.ms/vs/17/release/vc_redist.x86.exe if I've already done so once before?
What about the registration of the fpDotNet.dll? Isn't that a requirement?

I believe you that it work for you. I just don't understand why it such a pain for me. I was so looking forward to trying this out as a replacement for the older IE ActiveX.

**Comment by dsdsdfsdfsfsdfsfsdf** [comment]
Sorry registration fpDotNet.dll is also required. If step 3 has already been performed, then there is no need to perform it again.

**Comment by msueping** [comment]
I give up. I've never had such a difficult time getting anything FoxPro related to run. I know that the dll is registered cause it appears on the form in design time. It just won't run without encountering that wwDotNetBridge error. 

BTW, I noticed this about the "bridge" object in the procedure cefSharpBrowser.BindToHost. It looks suspicious to me.

<img width="1145" alt="image" src="https://user-images.githubusercontent.com/67658122/155425322-217b5cf0-81b7-4760-82e3-3a2cd2bdd7eb.png">

**Comment by dsdsdfsdfsfsdfsfsdf** [comment]
See issue to: https://github.com/dotnet/fsharp/issues/5037

**Comment by msueping** [comment]
OK. So now what? I'm afraid the content in that link is just beyond me. Also, why is it that you don't encounter the same issue?

**Comment by dsdsdfsdfsfsdfsfsdf** [comment]
Last try: Delete the entire C: \temp\CefSharp directory and download fpCefSharp.v97.1.61.zip file again. After downloading to disk, remove the Unblock flag and then expand the contents. The problem is probably that the fpCefSharp.v97.1.61.zip file is protected and Window11 probably also protects its contents. It could work after unblocking. 
![image](https://user-images.githubusercontent.com/99869018/155548851-06f48180-6acd-4e7d-80ef-6a4768ace46b.png)

**Comment by msueping** [comment]
Finally. It is exactly as you've pointed out above. The Unblock setting must be on in order for it to work correctly. At least it is in my scenario. Thanks for all of your input.
